site/app/controllers/calendar/CalendarController.php

```diff
@@ -103,27 +76,50 @@ public function createMessage(): RedirectResponse {
         }
         catch (\InvalidArgumentException $e) {
             $this->core->addErrorMessage($e->getMessage());
-            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+            return new RedirectResponse($this->core->buildUrl(['calendar']));
         }
         $calendar_item->setDate(new \DateTime($date));
         try {
             $calendar_item->setText($text);
         }
         catch (\InvalidArgumentException $e) {
             $this->core->addErrorMessage($e->getMessage());
-            return new RedirectResponse($this->core->buildCourseUrl(['calendar']));
+            return new RedirectResponse($this->core->buildUrl(['calendar']));
         }
 
-        $this->core->getCourseEntityManager()->persist($calendar_item);
-        $this->core->getCourseEntityManager()->flush();
+        if (isset($_POST['course'])) {
+            $set_course = $_POST['course'];
+        }
+        else {
+            $this->core->addErrorMessage("Invalid course given.");
+            return new RedirectResponse($this->core->buildUrl(['calendar']));
+        }
+
+        $instructor_courses = $this->core->getQueries()->getInstructorLevelUnarchivedCourses($this->core->getUser()->getId());
+        $exists = false;
+        foreach ($instructor_courses as $course) {
+            if ($set_course === ($course['semester'] . $course['course'])) {
```

Comment:
It would be better to check semester and course name each on their own. Here's an example where this could cause issues as is. Let's say there is are two semesters, one with code 's2' and another with code 's22'. Then there is also courses '2' and '22' in each semester. Instructors in semester 's2' course '22' could now create items on semester 's22' course '2'.

Reply:
To fix this issue, I had the $set_course incorporate a space between the semester and the course, I think this will improve readability when Instructors are specifying their course (In the problem you mention before if  professor was teaching "s2 22" and "s22 2", both would appear to read as "s222".) But this will also fix the issue you mention in your comment.

Reply:
It fixes issue because now the $set_course for these two would be different. One would be "s2 22" the other would be "s22 2"